### ISSUE TITLE
Allow user to send as a sub-address identity

### ISSUE THREAD

**Issue by jkrgr0** [mailcow/mailcow-dockerized#4871]
### Summary

When a user has enabled sub-addressing for its mailbox, the user is unable to send emails with this sub-address.  
Error message when trying to send an email with via SOGo with an additional identity for a sub-address:

```generic
5.7.1 <user+sub@domain.tld> Sender address rejected: not owned by user user@domain.tld
```

### Motivation

I use sub-addressing mainly for accounts on various online services. When I now need to communicate with the support of an online service via mail, they often complain that I am not sending from the email address that corresponds to the registered account.  

When allowing a user to send mails with a sub-address identity this issue won't be a thing.

**Comment by DerLinkman** [comment]
Sounds interesting. We'll look at this.

**Comment by FreddleSpl0it** [comment]
I don't think that this is possible with subaddressing. what you are looking for are aliases.

**Comment by MrHappy** [comment]
I use Evolution on Linux to accomplish this. Would be great to be able to do this on SOGo as well.

**Comment by MrHappy** [comment]
@DerLinkman Any news on this? 
little bump 'cause I'm still very interested in having this functionality...

**Comment by MrHappy** [comment]
Any new/progress on this?
Now I can only reply using a sub-address (<mailbox>+<submail>@domain.com) from my laptop (using Evolution in my case). Would be great to be able to do this from my Android-client (Nine) or webui.

**Comment by pedrostefanogv** [comment]
Any updates on this feature?
It would be interesting in email responses to reply with the sub-address

**Comment by m4teh** [comment]
> ### Summary
> When a user has enabled sub-addressing for its mailbox, the user is unable to send emails with this sub-address. Error message when trying to send an email with via SOGo with an additional identity for a sub-address

Can you add the alias/identity in SoGo settings first and then send the email, as a work around? I recall possibly doing this in the past. See https://bugs.sogo.nu/view.php?id=4637

But it comes down to SoGo having to make it more streamlined and editable in their interface. 

> Any new/progress on this? Now I can only reply using a sub-address (+@domain.com) from my laptop (using Evolution in my case). Would be great to be able to do this from my Android-client (Nine) or webui.

Definitely can with FairEmail which imo is the best and absurdly feature rich open source Android client out there. It will automatically use your tagged address when replying if you enable the option and the email is tagged. Use it all the time, it's brilliant.

**Comment by CodeShellDev** [comment]
Implemented this using an override in `extra.cf` for postfix:

```
# used for sub addressing
smtpd_sender_login_maps =
    regexp:/opt/postfix/conf/custom_sub_addressing.regexp,
    proxy:mysql:/opt/postfix/conf/sql/mysql_virtual_sender_acl.cf
```

and `custom_sub_addressing.regexp`

```
# sub-addressing, normalizes john+amazon@example.com into john@example.com (outgoing)
/^([^+@]+)\+[^@]+@([^@]+)$/    ${1}@$2
```

Hope this helps someone!
